src/sync/pool.go

```diff
@@ -1,19 +1,25 @@
 package sync
 
-// Pool is a very simple implementation of sync.Pool. It does not actually
-// implement a pool.
+// Pool is a very simple implementation of sync.Pool.
 type Pool struct {
-	New func() interface{}
+	New   func() interface{}
+	items []interface{}
```

Comment:
Please add a `sync.Mutex` so this type is safe for concurrent use.

Reply:
The sync package documents not being safe to use from multiple threads

https://github.com/tinygo-org/tinygo/blob/release/src/sync/doc.go#L2

And we can see the mutex implementation is not really locking for concurrent use

https://github.com/tinygo-org/tinygo/blob/release/src/sync/mutex.go#L24

So I guess it's better to leave out `sync.Mutex` to better reflect the reality that it's not safe?

Reply:
@aykevl Do you want to weight in with an opinion here?

Reply:
Hmm, not sure about this one to be honest.
_Right now_ we don't support parallelism and there are no blocking operations in this code so the code is safe. Once we add parallelism this code will become unsafe, along with many other parts of TinyGo internal code (like channels).

It does seem reasonable to me to use a mutex here, just to make sure this part of the code is forward compatible. But I would also be fine leaving it as-is, as adding parallelism will require reviewing lots of concurrency related code anyway.

Sidenote: concurrency and parallism aren't the same thing. TinyGo fully support concurrency (multiple goroutines). It doesn't support parallelism (goroutines running _at the same time_ on different CPU cores).